tokio-trace-fmt/src/filter.rs

```diff
@@ -59,9 +60,14 @@ impl EnvFilter {
             .cloned()
             .unwrap_or(Level::ERROR);
 
+        let includes_span_directive = directives
+            .iter()
+            .fold(false, |_, directive| directive.in_span.is_some());
```

Comment:
I think that rather than a `fold`, this could just be
```suggestion
            .any(|directive| directive.in_span.is_some());
```
This is simpler, and it has the advantage of being potentially more efficient (as `Iterator::any` will short-circuit when it finds an item matching the predicate). Admittedly, this isn't in the hot path so that efficiency is not terribly important.

Reply:
Also, TIOLI, but I think the _most_ efficient thing to do would be to just find the max level _and_ check whether any  directives include spans in one iteration over the directives, using a `for` loop:
```rust
let mut includes_spans = false;
let mut max_level = Level::ERROR;
for directive in directives {
    max_level = cmp::max(max_level, directive.level.clone());
    if directive.in_span.is_some() {
        includes_span_directives = true;
    }
}
```
On the other hand, using iterators might be more idiomatic. And again, this method isn't in a hot path (and the number of directives is unlikely to be so large that this actually matters). Just a thought, up to you.

Reply:
I find that code a bit hard to grok, I think I like the iterators and any a lot better personally.

Reply:
👍 that's fine w/ me